astropy/units/format/generic.py

```diff
@@ -457,6 +460,8 @@ def _parse_unit(cls, s: str, detailed_exception: bool = True) -> UnitBase:
         elif not s.isascii():
             if s[0] == "\N{MICRO SIGN}":
                 s = "u" + s[1:]
+            elif s[0] == "°":
```

Comment:
With the change in what is recognized as units, conversion of degree sign can now be done here.